kDriveCore/Data/Cache/DriveFileManager/DriveFileManager+Query.swift

```diff
@@ -0,0 +1,126 @@
+/*
+ Infomaniak kDrive - iOS App
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import RealmSwift
+
+public extension DriveFileManager {
+    func getCachedRootFile(freeze: Bool = true) -> File {
+        var file: File!
+        try? writeTransaction { writableRealm in
+            file = getCachedRootFile(freeze: freeze, writableRealm: writableRealm)
+        }
+        return file
+    }
+
+    func getCachedRootFile(freeze: Bool = true, writableRealm: Realm) -> File {
+        if let root = getCachedFile(id: DriveFileManager.constants.rootID, freeze: false, using: writableRealm) {
+            if root.name != drive.name {
+                root.name = drive.name
+            }
+            return freeze ? root.freeze() : root
+        } else {
+            return File(id: DriveFileManager.constants.rootID, name: drive.name, driveId: drive.id)
+        }
+    }
+
+    func getCachedMyFilesRoot() -> File? {
+        let file = try? fetchObject(ofType: File.self) { faultedCollection in
+            faultedCollection.filter("rawVisibility == %@", FileVisibility.isPrivateSpace.rawValue)
+                .first?
+                .freeze()
+        }
+
+        guard let file, !file.isInvalidated else {
+            return nil
+        }
+
+        return file
+    }
+
+    func getCachedFile(id: Int, freeze: Bool = true) -> File? {
+        let uid = File.uid(driveId: drive.id, fileId: id)
+        guard let file = fetchObject(ofType: File.self, forPrimaryKey: uid), !file.isInvalidated else {
+            return nil
+        }
+        return freeze ? file.freeze() : file
+    }
+
+    func getCachedFile(id: Int, freeze: Bool = true, using realm: Realm) -> File? {
+        let uid = File.uid(driveId: drive.id, fileId: id)
+        guard let file = realm.object(ofType: File.self, forPrimaryKey: uid), !file.isInvalidated else {
```

Comment:
Here it's okay to check if file is invalidated because realm could be live and could have changed

Reply:
okay